File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int, int> mp;

        for(int num : nums){
            mp[num]++;
        }

        int maxNum = 0;

        for(const auto& [key, val] : mp){
            maxNum = max(maxNum, val);
        }

        return maxNum > 1;
    }
};

File: Data Structures & Algorithms/binary-search/submission-0.java
class Solution {
    public int search(int[] nums, int target) {
        int min = 0;
        int mid;
        int max = nums.length - 1;

        while (max >= min) {
            mid = min + (max - min) / 2;
            if (nums[mid] == target) {
                return mid;
            }

            if (nums[mid] < target) {
                min = mid + 1;
            } else {
                max = mid - 1;
            }
        }
        return -1;
    }
}
